various: unpin old GCCs

emilazy opened the pull request:
Easy cases on the path towards deleting a whole bunch of EOL compilers.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux *(a few)*
  - [x] aarch64-linux *(the rest)*
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilazy committed:
directx-shader-compiler: unpin GCC

Seems to build fine now.

emilazy committed:
bazel_6: unpin GCC

emilazy committed:
chow-tape-model: patch for GCC ≥ 12

emilazy committed:
texlive: unpin GCC

Seems to build fine with the default now.

emilazy committed:
falcon: unstable-2018-10-23 -> 0-unstable-2023-11-19

Fixes the build with the default GCC.

emilazy committed:
llvmPackages_13.*: unpin GCC

This was added to LLVM 6, 8, 9, 10, 13, and 14 (but no others) in
695fe131c264a3cc244c9b07c71335f07bb92452, and then I guess gradually
got removed from all of them except 13. Anyway, I successfully built
`llvmPackages_13.stdenv` with the default GCC.

emilazy committed:
python313Packages.itk: bump GCC to 13

emilazy committed:
dmd: unpin GCC

Seems to build fine with the default GCC.

wolfgangwalther reviewed (commented):
Some build failures. Might be more than that, the builds are still running.

magnetophon reviewed (commented):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 434139`
Commit: `9aa7d8f3928c478b891faa597c2923dba823eaa6`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>chow-tape-model</li>
  </ul>
</details>
<details>
  <summary>:x: 92 packages failed to build:</summary>
  <ul>
    <li>anytype</li>
    <li>anytype-heart</li>
    <li>bant</li>
    <li>bazel (bazel_6)</li>
    <li>clang_13 (llvmPackages_13.clang, llvmPackages_13.libstdcxxClang)</li>
    <li>cone</li>
    <li>dale</li>
    <li>dmd</li>
    <li>driversi686Linux.amdvlk</li>
    <li>envoy</li>
    <li>falcon</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-mozc-ut</li>
    <li>mozc (ibus-engines.mozc)</li>
    <li>mozc-ut (ibus-engines.mozc-ut)</li>
    <li>klee</li>
    <li>libsForQt5.kdev-php (plasma5Packages.kdev-php)</li>
    <li>libsForQt5.kdev-python (plasma5Packages.kdev-python)</li>
    <li>libsForQt5.kdevelop (plasma5Packages.kdevelop)</li>
    <li>libsForQt5.kdevelop-unwrapped (plasma5Packages.kdevelop-unwrapped)</li>
    <li>lld_13 (llvmPackages_13.lld)</li>
    <li>lld_13.dev (llvmPackages_13.lld.dev)</li>
    <li>lld_13.lib (llvmPackages_13.lld.lib)</li>
    <li>llvmPackages_13.bintools</li>
    <li>llvmPackages_13.bintools-unwrapped</li>
    <li>llvmPackages_13.bintoolsNoLibc</li>
    <li>llvmPackages_13.clang-manpages</li>
    <li>llvmPackages_13.clang-manpages.debug</li>
    <li>llvmPackages_13.clang-tools</li>
    <li>llvmPackages_13.libclang (llvmPackages_13.clang-unwrapped)</li>
    <li>llvmPackages_13.libclang.debug (llvmPackages_13.clang-unwrapped.debug)</li>
    <li>llvmPackages_13.libclang.dev (llvmPackages_13.clang-unwrapped.dev)</li>
    <li>llvmPackages_13.libclang.lib (llvmPackages_13.clang-unwrapped.lib)</li>
    <li>llvmPackages_13.libclang.python (llvmPackages_13.clang-unwrapped.python)</li>
    <li>llvmPackages_13.clangNoLibcNoRt (llvmPackages_13.clangNoCompilerRt)</li>
    <li>llvmPackages_13.clangNoCompilerRtWithLibc</li>
    <li>llvmPackages_13.clangNoLibc (llvmPackages_13.clangNoLibcWithBasicRt)</li>
    <li>llvmPackages_13.clangNoLibcxx (llvmPackages_13.clangWithLibcAndBasicRt)</li>
    <li>llvmPackages_13.clangUseLLVM</li>
    <li>llvmPackages_13.clangWithLibcAndBasicRtAndLibcxx</li>
    <li>llvmPackages_13.compiler-rt (llvmPackages_13.compiler-rt-libc)</li>
    <li>llvmPackages_13.compiler-rt.dev (llvmPackages_13.compiler-rt-libc.dev)</li>
    <li>llvmPackages_13.compiler-rt-no-libc</li>
    <li>llvmPackages_13.compiler-rt-no-libc.dev</li>
    <li>llvmPackages_13.libcxx</li>
    <li>llvmPackages_13.libcxx.dev</li>
    <li>llvmPackages_13.libcxxClang</li>
    <li>llvmPackages_13.libcxxStdenv</li>
    <li>llvmPackages_13.libunwind</li>
    <li>llvmPackages_13.libunwind.dev</li>
    <li>llvmPackages_13.openmp</li>
    <li>llvmPackages_13.stdenv</li>
    <li>mathematica</li>
    <li>mathematica-cuda</li>
    <li>mathematica-webdoc</li>
    <li>mathematica-webdoc-cuda</li>
    <li>obliv-c</li>
    <li>oclgrind</li>
    <li>perf_data_converter</li>
    <li>pomerium</li>
    <li>protoc-gen-js</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.flowjax</li>
    <li>python312Packages.flowjax.dist</li>
    <li>python312Packages.jaxlib-build</li>
    <li>python312Packages.jaxlib-build.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python313Packages.jaxlib-build</li>
    <li>python313Packages.jaxlib-build.dist</li>
    <li>tests.cc-wrapper.llvmTests.llvmPackages_13.clang</li>
    <li>tests.cc-wrapper.llvmTests.llvmPackages_13.libcxx</li>
    <li>tests.cc-wrapper.supported</li>
    <li>verible</li>
    <li>veridian</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>bfc</li>
    <li>cling</li>
    <li>llvm_13 (llvmPackages_13.llvm)</li>
    <li>llvm_13.dev (llvmPackages_13.llvm.dev)</li>
    <li>llvm_13.lib (llvmPackages_13.llvm.lib)</li>
    <li>llvm_13.python (llvmPackages_13.llvm.python)</li>
    <li>llvmPackages_13.llvm-manpages</li>
    <li>llvmPackages_13.tblgen</li>
    <li>python312Packages.itk</li>
    <li>python313Packages.itk</li>
    <li>xeus-cling</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>bazel</summary>
<pre>./scripts/ci/ci.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/ci/build_status_command.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/ci/build.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/bootstrap/buildenv.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/bootstrap/bootstrap.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/bazel-dev.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/bazel-bisect.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./scripts/bash_completion_test.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./examples/shell/test.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./examples/shell/lib.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./examples/shell/bin.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./compile.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/b4jl9k37x6hrb2qb7n63dsrfiw0ywmjq-bash/bin/bash"
./combine_distfiles_to_tar.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/lb33m49aslmvkx5l4xrkiy7m6nbh2kqf-bash-interactive-5.3p0/bin/sh"
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: preBuildPhase
Running phase: buildPhase
Read from remote host 81.206.32.45: Connection reset by peer
client_loop: send disconnect: Broken pipe</pre>
</details>
<details>
<summary>dale</summary>
<pre>Read from remote host 81.206.32.45: Connection reset by peer
client_loop: send disconnect: Broken pipe</pre>
</details>
<details>
<summary>dmd</summary>
<pre>dmd/compiler/test/compilable/testclidflags.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/compilable/testcolor.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/compilable/extra-files/ddocAny-postscript.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/compilable/extra-files/ddocYear-postscript.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/compilable/extra-files/json-postscript.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/compilable/extra-files/vcg-ast-arraylength-postscript.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/tools/postscript.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
dmd/compiler/test/tools/sh_do_test.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash"
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
(TX) VERSION
(TX) SYSCONFDIR
(DC) COMMON
(TX) DMD_CONF
(DC) LEXER
(DC) BACKEND
Read from remote host 81.206.32.45: Connection reset by peer
client_loop: send disconnect: Broken pipe</pre>
</details>
<details>
<summary>falcon</summary>
<pre>  500 |       register int64 ret = loadInt64(
      |                      ^~~
/build/source/include/falcon/vm.h: In member function 'Falcon::int32 Falcon::VMachine::getNextNTD32()':
/build/source/include/falcon/vm.h:491:22: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  491 |       register int32 ret = *reinterpret_cast<int32 *>(
      |                      ^~~
/build/source/include/falcon/vm.h: In member function 'Falcon::int64 Falcon::VMachine::getNextNTD64()':
/build/source/include/falcon/vm.h:500:22: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  500 |       register int64 ret = loadInt64(
      |                      ^~~
[ 75%] Built target sqlite3_fm
[ 75%] Linking CXX executable ../../bin/falpack
[ 75%] Built target falpack
[ 75%] Linking CXX shared module ../../../../lib/falcon/dbi_fm.so
[ 75%] Built target dbi_fm
[ 76%] Linking CXX shared module ../../../../lib/falcon/hash_fm.so
[ 76%] Built target hash_fm
[ 77%] Linking CXX shared module ../../../../lib/falcon/bufext_fm.so
[ 77%] Built target bufext_fm
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>llvmPackages_13.compiler-rt</summary>
<pre>copying path '/nix/store/wfbzv6p35wfg394qapwq27rc5k71ppix-armv6-sync-ops-no-thumb.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/r7rxxipc6nigdp848yi639g3dkx6d6d4-armv6-scudo-libatomic.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/xhc43z7x1wmx50j4a43i27vkyfp9fzf7-source' from 'https://cache.nixos.org'...
copying path '/nix/store/x0137mdpq76zfp2xvn3ncawys2g4c93i-normalize-var.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/rd6gan216l9rm9xs5whbkzsc6c6kqq31-armv6-mcr-dmb.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/0ncx0876rq0gx77k1paxw5cjvsxkg9v9-armv6-scudo-no-yield.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/jgqk8bhzzvwb8d3yjlmyk319mnrrphcg-armv7l.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/88ccn8m0kpg0cqjzbpl5hymn6z3lqb7w-cfi_startproc-after-label.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/pmw5l118lmhak47aqlz71a87xgaw3alb-X86-support-extension.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/dbapbg570sx4xsbp50s52azwh6agj40m-darwin-targetconditionals.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/5pi27cr43baazlvy239qdj4c1jzzbfv7-codesign.patch' from 'https://cache.nixos.org'...
Read from remote host 81.206.32.45: Connection reset by peer
client_loop: send disconnect: Broken pipe</pre>
</details>
<details>
<summary>obliv-c</summary>
<pre>gcc -c src/ext/oblivc/commitReveal.c -o _build/commitReveal.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_network_utils.c -o _build/obliv_network_utils.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/bcrandom.c -o _build/bcrandom.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/privacy-free.c -o _build/privacy-free.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/psi.c -o _build/psi.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/nnob.c -o _build/nnob.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/copy.c -o _build/copy.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_add.c -o _build/obliv_float_add.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_sub.c -o _build/obliv_float_sub.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_div.c -o _build/obliv_float_div.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_eq.c -o _build/obliv_float_eq.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_le.c -o _build/obliv_float_le.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_lt.c -o _build/obliv_float_lt.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_mult.c -o _build/obliv_float_mult.o -I src/ext/oblivc -g -O2 -O3
gcc -c src/ext/oblivc/obliv_float_neg.c -o _build/obliv_float_neg.o -I src/ext/oblivc -g -O2 -O3
bin/oblivcc -c src/ext/oblivc/copy.oc -o _build/copy.oo -I src/ext/oblivc -g -O2 -O3
/nix/store/74qjr01q87nwfl0dbsr1s45p8crw3q1f-glibc-2.40-66-dev/include/stdlib.h:697:12: error: 'reallocarray' undeclared here (not in a function)
  697 |      __attribute_alloc_size__ ((2, 3))
      |            ^~~~~~~~~~~~
make: *** [_build/depends/copy.od:5: _build/copy.oo] Error 1</pre>
</details>
</details>

@emilazy Related, could you have a look at:
https://github.com/NixOS/nixpkgs/pull/431533

emilazy commented:
I’ve fixed `falcon` with a bump (it was already broken on Darwin and I don’t feel like putting more effort into a package that is evidently undermaintained both upstream and downstream, so I’ve just marked it broken there). `obliv-c` is not an easy case after all, so I have moved it out of this PR. `libsForQt5.kdev-python` is a preexisting failure (and due for removal anyway), `mathematica` is presumably because of the `requireFile`, and `driversi686Linux.amdvlk` looks like you just ran out of memory?

emilazy commented:
So, uh, going by https://github.com/GPUOpen-Drivers/gpurt/issues/5 it sort of looks like some kind of cursed UB in `directx-shader-compiler` that only happens on 32 bit and that the old GCC masks? Very cool. I’m going to see if disabling LTO fixes it or something.

emilazy committed:
amdvlk: disable ray tracing support for `i686-linux`

emilazy commented:
Yeah, it’s pretty cursed. At the risk of going outside the bounds of easy cases, I have followed the lead of Arch and simply disabled ray tracing on 32‐bit AMDVLK – they turned it off years ago, probably in response to this exact issue, and nobody seems to have noticed. Presumably every game that uses ray tracing is 64 bit anyway.

Flakebi commented:
The amdvlk change sounds ok to me :)

emilazy commented:
Thanks for the reviews! I will be more thorough about testing the upcoming PRs.
